### DISCUSSION TITLE
Existing Fiat accounting system + BTCPay (Real World Problem)

### DISCUSSION THREAD

**Discussion by 0xjacobb** [btcpayserver/btcpayserver#3010] [Technical Support] [answered]
Hi,

**Situation:**
I write invoices with my existing Fiat account system. Invoices are based on a template. On this template I have a fix URL to my BTCPay server, for customers who want to pay the invoice with BTC. Various customers with various prices.

**Problem:**
I need a setting, where my back-office could link a BTC payment to a customer or specific invoice. Actually I use the POS app on BTCPay for that, but there is no "ID-field" or "customer name field" which could be used for linking the BTC payment to a customer/invoice from my accounting system.

**My setup:**
- Existing fiat accounting system
- BTCPay on RPI4
- POS app on BTCPay

**GOAL:**
A setup where we still could use our existing Fiat accounting system and if a customer pay with BTC my back-office could link this payment to this invoice. 

Sending a specific invoice or payment request directly from BTCPay is no option, because then my back-office has to set up two invoices on two different systems.

--> What is the best setup for this real world use case?

Thanks!

**Comment by pavlenex** [comment, marked as answer]
@0xjacobb Do you use point of sale in retail (brick and mortar store)?

Easy way to link these would be to enable email `refund address` in Checkout experience. Though I am not sure if that would make much sense and if your customers provide email.

**Comment by 0xjacobb** [comment]
@pavlenex What do you mean with `Do you use point of sale in retail (brick and mortar store)?` --> Yes I use actually the PoS for my setup, but don't have a store or "retail PoS system". I provide consultancy and don't have a "classic retail store". In my sense, it was the easiest way to provide an easy UX for my customers to pay in BTC.

How would you set up this checkout experience?

**Reply by pavlenex** [reply]
So you're using PoS as a web-store, which to be honest isn't it's main purpose. People in retail don't have a way to identify their customers in this way.
You can enable email request 
![Screenshot 2021-10-26 at 09 21 52](https://user-images.githubusercontent.com/36959754/138828139-fe9f2eef-818b-48f5-b7d3-6a5247b9c9d5.png)
in Store Settings > Checkout experience.
This would enable email prompt which will be added in the invoice fields.

**Reply by 0xjacobb** [reply]
Yes I know, that the PoS is not the real use case for that. But it was the easiest way IMHO. 
--< How would you setup the BTCPay to solve my problem?

**Reply by pavlenex** [reply]
Since you use it for online purpose, as I said I would enable email collection when invoice is generated, and you can do that with the settings I showed above.

Besides `PoS` you can use [payment request](https://docs.btcpayserver.org/PaymentRequests/#payment-requests) from the header they're basically a way to invoice your clients, but you'd have to fill out info manually for every client (similar to making an invoice in the fiat accounting).

**Reply by 0xjacobb** [reply]
ok thanks. Just for understanding: What would be your setup, if you have a FIAT accounting system which should implement BTCPay payments easily (like my described use case)? 

Yes, I've experimented with `payment requests`, but as you said, it is an additional manual step in the process. Would be nice if there is an app function which combines the PoS with an additional ID for the accounting system.

**Reply by pavlenex** [reply]
Well you're using something that's not meant to be used that way, but it works for you so I would enable `refund email`. If you're looking for an advanced way to invoice clients, use payment requests, that's what they are for. You can always export invoices and add it in your fiat accounting in both cases.

**Reply by 0xjacobb** [reply]
ok. Is there an automated way (webhooks or similar) where payment request could be automated?

**Reply by pavlenex** [reply]
I don't know you should check the greenfield api docs for this. https://docs.btcpayserver.org/API/Greenfield/v1/

**Reply by 0xjacobb** [reply]
I will. Thanks for your help
